### PR TITLE
Adding integration tests for iterators

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This adds integration tests verifying the correctness of various ways of iterating over `Results` (unfiltered & filtered), `lists` and `linkingObjects`.

## ☑️ ToDos
* [ ] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: integration-tests/tests/src/iterators.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            realm.objects("SomeOtherClass");
+        }).throws("Object type 'SomeOtherClass' not found in schema.");
+    });
+
+    type CollectionCallback = () => Realm.Collection<any>;
+
+    function itCanIterate(
+        getCollection: CollectionCallback,
+        expectedNames: string[]
+    ) {
```
Comment (blagoev) [inline integration-tests/tests/src/iterators.ts]: function itCanIterate(
        getCollection: CollectionCallback,
        expectedNames: string[]) 
{

little and tiny code formatting note :)

Comment (kraenhansen) [reply integration-tests/tests/src/iterators.ts]: Not with the current `tslint` rules :)

Comment (blagoev) [reply integration-tests/tests/src/iterators.ts]: that sucks that our rules require us to do that.

Comment (kraenhansen) [reply integration-tests/tests/src/iterators.ts]: Personally I don't really care either way, as long as we have a standard that I can use automated tools to format my code into. Our [tslint.json](https://github.com/realm/realm-js/blob/master/integration-tests/tests/tslint.json) mostly use the recommended / latest options, in an effort to make us less special compared to other code-bases on the interwebs.

**Pull request merged**